queue_services/business-events-listener/src/business_events_listener/worker.py

```diff
@@ -64,10 +64,58 @@ async def process_event(event_message, flask_app):
     with flask_app.app_context():
         message_type = event_message.get('type', None)
 
-        if message_type == 'bc.registry.names.events':
+        if message_type in ('bc.registry.business.dissolution',
+                            'bc.registry.business.correction'):
+            await process_business_events(message_type, event_message)
+        elif message_type == 'bc.registry.names.events':
             await process_name_events(event_message)
 
 
+async def process_business_events(message_type: str, event_message: Dict[str, any]):
+    """Process business events.
+
+    1. Check if entity exists and dissolution has occured.
+       - Update the status to HISTORICAL.
+    2. **Unimplemented ** Check if entity exists and dissolution has reversed (via correction).
+       - Update the status to ACTIVE.
+
+    Args:
+        event_message (object): cloud event message, sample below.
+            {
+                'specversion': '1.x-wip',
+                'type': 'bc.registry.business.' + filing.filing_type,
+                'source': '/business/{business.identifier}/filing/{filing.id},
+                'id': id,
+                'time': '',
+                'datacontenttype': 'application/json',
+                'identifier': business.identifier,
+                'data': {
+                    'filing': {
+                        'header': {
+                                   'filingId': filing.id,
+                                   'effectiveDate': filing.effective_date
+                                  },
+                        'business': {'identifier': business.identifier},
+                        'legalFilings': get_filing_types(filing.filing_json)
+                    }
+                }
+            }
+    """
+    logger.debug('>>>>>>>process_business_events>>>>>')
+    entity_identifier = event_message.get('identifier')
+    entity = EntityModel.find_by_business_identifier(entity_identifier)
+    if entity is None:
+        logger.warn('process_business_events - Couldn''t find Entity : '
+                    f'{entity_identifier} to update entity.status.')
+    elif message_type == 'bc.registry.business.dissolution':
+        logger.info(f'Setting Entity :{entity_identifier} status to HISTORICAL')
+        entity.status = EntityStatus.HISTORICAL
+        entity.save()
+    elif message_type == 'bc.registry.business.correction':
```

Comment:
In the future will need to check to see if it's currently in HISTORICAL. 
Also I noticed there is no easy way of mapping modified_by. LEAR has it's own submitter_id which is an integer:
`lear.submitter_id -> lear.keycloak-id -> auth_db.modified_by_id`